synapse/rest/media/v1/media_storage.py

```diff
@@ -112,14 +116,20 @@ def store_into_file(self, file_info: FileInfo):
 
         finished_called = [False]
 
-        async def finish():
-            for provider in self.storage_providers:
-                await provider.store_file(path, file_info)
-
-            finished_called[0] = True
-
         try:
             with open(fname, "wb") as f:
```

Comment:
Huh, this being a context manager here is a bit weird when we don't really write inside of it...

Reply:
Well, we do in that we write during the `yield`, as the function is a context manager?